src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-snapshot-list/rbd-snapshot-list.component.ts

```diff
@@ -0,0 +1,279 @@
+} from '../../../shared/components/delete-confirmation-modal/delete-confirmation-modal.component';
+import { CdTableColumn } from '../../../shared/models/cd-table-column';
+import { CdTableSelection } from '../../../shared/models/cd-table-selection';
+import { ExecutingTask } from '../../../shared/models/executing-task';
+import { FinishedTask } from '../../../shared/models/finished-task';
+import { CdDatePipe } from '../../../shared/pipes/cd-date.pipe';
+import { DimlessBinaryPipe } from '../../../shared/pipes/dimless-binary.pipe';
+import {
+  NotificationService
+} from '../../../shared/services/notification.service';
+import { TaskManagerService } from '../../../shared/services/task-manager.service';
+import { RbdSnapshotFormComponent } from '../rbd-snapshot-form/rbd-snapshot-form.component';
+import {
+  RollbackConfirmationModalComponent
+} from '../rollback-confirmation-modal/rollback-confimation-modal.component';
+import { RbdSnapshotModel } from './rbd-snapshot.model';
+
+@Component({
+  selector: 'cd-rbd-snapshot-list',
+  templateUrl: './rbd-snapshot-list.component.html',
+  styleUrls: ['./rbd-snapshot-list.component.scss']
+})
+export class RbdSnapshotListComponent implements OnInit, OnChanges {
+
+  @Input() snapshots: RbdSnapshotModel[] = [];
+  @Input() poolName: string;
+  @Input() rbdName: string;
+  @Input() executingTasks: ExecutingTask[] = [];
+
+  @ViewChild('nameTpl') nameTpl: TemplateRef<any>;
+  @ViewChild('protectTpl') protectTpl: TemplateRef<any>;
+
+  data: RbdSnapshotModel[];
+
+  columns: CdTableColumn[];
+
+  modalRef: BsModalRef;
+
+  selection = new CdTableSelection();
+
+  constructor(private modalService: BsModalService,
+              private dimlessBinaryPipe: DimlessBinaryPipe,
+              private cdDatePipe: CdDatePipe,
+              private rbdService: RbdService,
+              private toastr: ToastsManager,
+              private taskManagerService: TaskManagerService,
+              private notificationService: NotificationService) { }
+
+  ngOnInit() {
+    this.columns = [
+      {
+        name: 'Name',
+        prop: 'name',
+        cellTemplate: this.nameTpl,
+        flexGrow: 2
+      },
+      {
+        name: 'Size',
+        prop: 'size',
+        flexGrow: 1,
+        cellClass: 'text-right',
+        pipe: this.dimlessBinaryPipe
+      },
+      {
+        name: 'Provisioned',
+        prop: 'disk_usage',
+        flexGrow: 1,
+        cellClass: 'text-right',
+        pipe: this.dimlessBinaryPipe
+      },
+      {
+        name: 'State',
+        prop: 'is_protected',
+        flexGrow: 1,
+        cellClass: 'text-center',
+        cellTemplate: this.protectTpl
+      },
+      {
+        name: 'Created',
+        prop: 'timestamp',
+        flexGrow: 1,
+        pipe: this.cdDatePipe
+      }
+    ];
+  }
+
+  ngOnChanges() {
+    this.data = this._merge(this.snapshots, this.executingTasks);
+  }
+
+  _merge(snapshots: RbdSnapshotModel[], executingTasks: ExecutingTask[] = []) {
```

Comment:
If this function should be private then consider using ``private merge(...)`` here.